src/lib.rs

```diff
@@ -9,30 +9,33 @@
 // except according to those terms.
 
 #![no_std]
-#![cfg_attr(feature = "alloc", feature(alloc))]
+
+#![cfg_attr(feature = "nightly", feature(cfg_target_has_atomic))]
 
 #[cfg(test)]
 #[macro_use]
 extern crate std;
 
-#[cfg(feature = "alloc")]
 extern crate alloc;
 extern crate cty;
 extern crate memchr;
 
-#[cfg(feature = "alloc")]
+// If building with the nightly toolchain and "nightly" feature, we can optionally
+// check if our target actually supports atomic operations.
+// If building with stable, we will include Arc, but the build may fail here for some targets.
+#[cfg(any(
+        not(feature="nightly"),
+        all(feature = "nightly", target_has_atomic = "ptr")
+         )
+     )
+]
```

Comment:
Actually I think there is a better way to handle this: just add an `arc` feature that is enabled by default. That way users on thumbv6 targets can explicitly disable the feature.